cmd/codegen/main.go

```diff
@@ -112,18 +116,43 @@ func ClientGen(cmd *cobra.Command, args []string) error {
 		cfg.IntrospectionJSON = string(introspectionJSON)
 	}
 
-	if dependenciesJSONFilePath != "" {
-		dependenciesJSON, err := os.ReadFile(dependenciesJSONFilePath)
+	if moduleSourceID != "" {
+		dag, err := dagger.Connect(ctx)
 		if err != nil {
-			return fmt.Errorf("read dependencies json: %w", err)
+			return fmt.Errorf("failed to connect to engine: %w", err)
 		}
+		defer dag.Close()
 
-		var gitDependencies []modules.ModuleConfigDependency
-		if err := json.Unmarshal(dependenciesJSON, &gitDependencies); err != nil {
-			return fmt.Errorf("failed to unmarshal dependencies json: %w", err)
+		var res struct {
+			Source struct {
+				Dependencies []generator.ModuleSourceDependencies
+			}
+		}
+
+		err = dag.Do(ctx,
+			&dagger.Request{
+				Query:  loadModuleSourceDepsQuery,
+				OpName: "ModuleSourceDependencies",
+				Variables: map[string]any{
+					"source": dagger.ModuleSourceID(moduleSourceID),
+				},
+			},
+			&dagger.Response{
+				Data: &res,
+			})
+		if err != nil {
+			return fmt.Errorf("failed to load module source dependencies: %w", err)
+		}
+
+		cfg.ModuleDependencies = res.Source.Dependencies
+
+		// If the module source is implementing functions, we need to load the module itself
+		src, err := dag.LoadModuleSourceFromID(dagger.ModuleSourceID(moduleSourceID)).SDK().Source(ctx)
```

Comment:
This is a new feature right? We weren't previously serving our own source back right?

I wonder if because we now have a ModuleSourceID, if we could now use the new `includeDependencies`, instead of doing the manual loading: https://github.com/dagger/dagger/blob/cc5e55e8bed125117b241d204714a4353810af6a/docs/docs-graphql/schema.graphqls#L2803-L2812

Reply:
I'm not sure I understood this comment.

The `Source` is a simple string field that point to where the SDK is.
It's only useful for the codegen to know if the module we generate for implements its own functions.

I'm not sure we can run `serve` on a module that has no source.

Reply:
> I'm not sure I understood this comment.

The comment is probably not on the right line.

I have two points:
- Were we previously serving + generating the API of our own module back to ourselves? Or is this new?
- If we *are* serving our own API, can we use the method I link to? Does that make sense? It feels like we're implementing lots of the same logic as we've already implemented there.

Reply:
It's new, in module we never generated the API of our own module.
I'm gonna check if we can serve an SDK less dependencies or not, because it might simplify a lot the resolution if we can, I could just add "include-dependencies" and boom

Reply:
It works! I'm using the include deps so I do not even need to serve all modules in a loop